src/test/java/DisplayTest.java

```diff
@@ -0,0 +1,61 @@
+import org.junit.*;
+
+import java.io.*;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertThat;
+
+public class DisplayTest {
+    private final ByteArrayOutputStream outputContent = new ByteArrayOutputStream();
+    private final PrintStream defaultOut = System.out;
+    private Display display;
+
+    @Before
+    public void setUpStreams() {
+        display = new Display();
+        System.setOut(new PrintStream(outputContent));
```

Comment:
Ah, I see how you've tested this. You're mutating global state here (and resetting it correctly so you've "managed" it, but I think a better way to do this is to inject your dependency into the class under test (CUT) so you don't have to think about global (static) state.

Reply:
✅ worked through understanding spies in Java and created a hand rolled spy. PR #7